src/aap_eda/api/serializers/decision_environment.py

```diff
@@ -70,6 +74,28 @@ class DecisionEnvironmentCreateSerializer(serializers.ModelSerializer):
             validators.check_credential_registry_username_password,
         ],
     )
+    pull_policy = serializers.CharField(
+        required=False,
+        default=settings.DEFAULT_PULL_POLICY.lower(),
+        help_text="Pull policy for the container image",
+    )
+
+    def validate_pull_policy(self, value):
+        """Validate and normalize pull policy input."""
+        valid_inputs = {
+            ImagePullPolicy.ALWAYS.value,
+            ImagePullPolicy.NEVER.value,
+            ImagePullPolicy.MISSING.value,
+            settings.DEFAULT_PULL_POLICY,
+        }
```

Comment:
The enum should be the source of truth, we should not hardcode the values here. 
`settings.DEFAULT_PULL_POLICY` must be validated to meet a valid enum, it should not be here. 
The open api spec should be honored.

```suggestion
        valid_inputs = {policy.value for policy in ImagePullPolicy}
```

Reply:
removed